packages/server/tests/acceptance/erc20.spec.ts

```diff
@@ -0,0 +1,297 @@
+                                        `CALL_EXCEPTION`,
+                                    );
+                                });
+                            });
+                        });
+
+                        describe('when the spender has unlimited allowance', function () {
+                            beforeEach(async function () {
+                                await contract.connect(tokenOwnerWallet).approve(spender, ethers.constants.MaxUint256);
+                            });
+
+                            it('does not decrease the spender allowance', async function () {
+                                await contract.connect(spenderWallet).transferFrom(tokenOwner, to, 1);
+                                const allowance = await contract.allowance(tokenOwner, spender);
+                                expect(allowance.toString()).to.be.equal(ethers.constants.MaxUint256.toString());
+                            });
+
+                            it('does not emit an approval event', async function () {
+                                await expect(await contract.connect(spenderWallet).transferFrom(tokenOwner, to, 1))
+                                    .to.not.emit(contract, 'Approval');
+                            });
+                        });
+                    });
+
+                    describe('when the recipient is the zero address', function () {
+                        let amount, to, tokenOwnerWallet;
+
+                        beforeEach(async function () {
+                            amount = initialSupply;
+                            to = ethers.constants.AddressZero;
+                            tokenOwnerWallet = accounts[0].wallet;
+                            await contract.connect(tokenOwnerWallet).approve(spender, amount);
+                        });
+
+                        it('reverts', async function () {
+                            await expectRevert(contract.connect(spenderWallet).transferFrom(tokenOwner, to, amount),
+                                `CALL_EXCEPTION`);
+                        });
+                    });
+                });
+            });
+        });
+    });
+
+    const expectRevert = async (promise, code) => {
+        const tx = await promise;
+        try {
+            await tx.wait();
+            Assertions.expectedError();
+        }
+        catch(e:any) {
+            expect(e).to.exist;
+            expect(e.code).to.eq(code);
+        }
+    };
+
+    const deployErc20 = async (constructorArgs:any[] = [], contractJson) => {
+        const factory = new ethers.ContractFactory(contractJson.abi, contractJson.bytecode, accounts[0].wallet);
+        let contract = await factory.deploy(...constructorArgs);
+
+        await contract.deployed();
+
+        // FIXME mirror node calls should not be made
+        const contractResult = await mirrorNode.get(`/contracts/results/${contract.deployTransaction.hash}`);
```

Comment:
When deploying through `ethers` the pre-deployed address is saved in the contract object. That address is actually incorrect, since it is not yet submitted to the consensus node. Unfortunatelly the `.deployed` method does not update the address, so the contract has to be re-initiated with the correct one.

Reply:
yes. it is logged as #193